core/storage/user/gae_models.py

```diff
@@ -1878,8 +1878,13 @@ class UserQueryModel(base_models.BaseModel):
 
     The id of each instance of this model is alphanumeric id of length 12
     unique to each model instance.
+
+    This model turns off caching, because this results in stale data being
+    shown after each UserQueryOneOffJob.
     """
 
+    _use_cache = False
+    _use_memcache = False
```

Comment:
Is this an issue you noticed now? Given that the PR is meant to be a refactor, I wouldn't expect there to be a behavioral regression. How did this issue manifest itself? I'm also a bit surprised given other jobs don't have this issue.

Reply:
How I found this issue and the issue:
This occurs in develop also with the email dashboard

First I create a user such that it satisfies a condition, 
Next I run userQuery job in the emaildashboard page

1) Try clicking re-check status before the actual job completes, the result is 0
2) Only click the re-check status button after confirming the job is complete in the terminal and then click it, the result is 1

Then when I disabled caching, the problem got fixed. This problem was never noticed because there was no e2e test for this part before and this was not a frequently used feature.